sonar-helm-for-iac/converters/data-loading.go

```diff
@@ -54,14 +58,25 @@ func PrepareChartValues(templateSources *TemplateSources) (map[string]interface{
 
 	result["Values"] = *values.MustGet()
 	result["Chart"] = *chart
-	result["Files"] = templateSources.files
 
 	result["Capabilities"] = DefaultCapabilities
 	result["Release"] = DefaultReleaseMetadata
-	result["Template"] = Template{
-		Name:     templateSources.Name,
-		BasePath: getBasePath(chart),
+
+	// Helm allows referencing files (e.g. in `.Files.*` functions or by name in `template` and `include`) with kind of fully-qualified path,
+	// which starts with chart name. See `pkg/chart/loader/load.go:LoadFiles`.
+	// TODO SONARIAC-1241: handle nested charts, i.e. basePath should incorporate parent path like "parent-chart/charts/nested-chart/templates"
+	chartPathPrefix := (*chart)["name"].(string)
+	for filename, content := range templateSources.files {
+		if strings.HasPrefix(filename, "templates/") {
+			delete(templateSources.files, filename)
+			templateSources.files[chartPathPrefix+"/"+filename] = content
```

Comment:
I don't like that `tesmplateSources` object is being mutated inside a function, but re-organizing everything would increase size of this PR. I'd rather do it in a separate one.